test/e2e-go/cli/goal/expect/goalDryrunRestTest.exp

```diff
@@ -21,6 +21,25 @@ proc TestGoalDryrun { DRREQ_FILE TEST_PRIMARY_NODE_DIR } {
     }
 }
 
+proc TestGoalDryrunFailure { DRREQ_FILE TEST_PRIMARY_NODE_DIR EXPECTED_STATUS_CODE EXPECTED_MESSAGE} {
+    set MESSAGE_MATCHED 0
+    spawn goal clerk dryrun-remote -d $TEST_PRIMARY_NODE_DIR -D $DRREQ_FILE -v
+    expect {
+        timeout { ::AlgorandGoal::Abort "goal clerk dryrun-remote timeout" }
+        $EXPECTED_MESSAGE {puts "message matched"; set MESSAGE_MATCHED 1; close}
+        eof
+    }
+    catch wait result
```

Comment:
does it really work outside if `expect` block?

Reply:
I believe so, the result variable does end up containing the exit code that we needed. Apparently this is how you wait for expect to hit eof

https://stackoverflow.com/questions/3299502/how-to-return-spawned-process-exit-code-in-expect-script

Reply:
I did not find it in the doc :(
To ensure, could you please check it by adding
```tcl
TestGoalDryrunFailure $DRREQ_FILE_3 $TEST_PRIMARY_NODE_DIR 0 "PASS"
```

Maybe `TestGoalDryrunFailure` might be renamed to `TestGoalDryrunExitCode`